source/adios2/operator/compress/CompressSZ.cpp

```diff
@@ -279,7 +279,7 @@ size_t CompressSZ::Compress(const void *dataIn, const Dims &dimensions,
     // In C, r[0] is the last dimension. In Fortran, r[0] is the first dimension
     for (int i = 0; i < ndims; i++)
     {
-        uint dsize = dimensions[i];
+        unsigned int dsize = dimensions[i];
```

Comment:
both dimensions and r are size_t arrays, so dsize should be too. Or, as the code stands now, this should be a single assignment.

Reply:
Well, I was being bug-for-bug compatible ;)

I updated the PR to get rid of `dsize` altogether, which is clearly the better solution.